### ISSUE TITLE
Quicklint Throws E0089 Error When Using Import Attribute Syntax for Bun’s Macros

### ISSUE THREAD

**Issue by weleo** [quick-lint/quick-lint-js#1232]
When utilizing the import attribute syntax as part of [Bun’s macros](https://bun.sh/docs/bundler/macros), Quicklint throws an [E0089 error](https://quick-lint-js.com/errors/E0089/). This issue arises because Quicklint currently expects with statements to enclose expressions within parentheses, which conflicts with the syntax introduced by the Stage 3 TC39 proposal for import attributes.

Example code causing the error:
```ts
import { macro } from './macro.js' with { type: 'macro' };
```

Quicklint should correctly parse and recognize the import attribute syntax without raising the E0089 error. The linter should accommodate the new syntax as defined in the [import attributes proposal](https://github.com/tc39/proposal-import-attributes).

<img width="938" alt="image" src="https://github.com/user-attachments/assets/da6582de-537a-4aee-a289-508654a1af7a" />

**Comment by strager** [comment]
You're correct that this is a bug in quick-lint-js. #1004 tracks adding support.
